terminal/bundles/org.eclipse.terminal.view.core/src/org/eclipse/terminal/view/core/ITerminalService.java

```diff
@@ -32,34 +33,48 @@ public interface Done {
 		public void done(IStatus status);
 	}
 
+	/**
+	 * Empty callback if no action is needed
+	 */
+	Done NOOP = new Done() {
+
+		@Override
+		public void done(IStatus status) {
+			//no-op
+		}
+	};
+
 	/**
 	 * Opens a terminal asynchronously and invokes the given callback if done.
 	 *
 	 * @param properties The terminal properties. Must not be <code>null</code>.
-	 * @param done The callback to invoke if finished or <code>null</code>.
+	 * @param done The callback to invoke if finished. Must not be <code>null</code>.
+	 * @see ITerminalService#NOOP
 	 */
 	public void openConsole(Map<String, Object> properties, Done done);
```

Comment:
```suggestion
	public void openConsole(Map<String, Object> properties, Done done);
	
	public default void openConsole(Map<String,Object> properties) {
	     openConsole(properties, s->{});
	}
```

and so on ...

Reply:
Thinking further about this returning a `CompletableFuture<IStatus>` might even be much better here ...